cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/SourceControlManager.java

```diff
@@ -95,20 +85,27 @@ public void initialise() throws GitAPIException, AuthenticationException, IOExce
       return;
     }
     // Clean up the directory if it already exists.
-    if (localRepoDirectory.exists() && !deleteDirectory(localRepoDirectory)) {
-      throw new IOException("Failed to clean up local repository directory " + context.getRepoBasePath());
-    }
-    String branch = "refs/heads/" + context.getRepositoryConfig().getDefaultBranch();
-    CloneCommand command = Git.cloneRepository()
-      .setURI(context.getRepositoryConfig().getDefaultBranch())
-      .setDirectory(localRepoDirectory)
-      .setBranchesToClone(Collections.singleton((branch)))
-      .setBranch(branch);
+    deleteIfRequired(localRepoDirectory);
+    CloneCommand command =
+      Git.cloneRepository().setURI(context.getRepositoryConfig().getDefaultBranch()).setDirectory(localRepoDirectory);
 
+    String branch = getBranchRefName(context.getRepositoryConfig().getDefaultBranch());
+    if (branch != null) {
+      command.setBranchesToClone(Collections.singleton((branch))).setBranch(branch);
```

Comment:
So if we don't set the branch, it's gonna be the default one from provider?

Reply:
Yes, the default is to checkout the HEAD of the remote repository. During the tests I observed that its the head of master brach. When testing pull, push I'll write tests with different remote HEADs.